src/compiler/checker.ts

```diff
@@ -5997,12 +5997,19 @@ namespace ts {
             }
         }
 
+
+
         // Get the narrowed type of a given symbol at a given location
         function getNarrowedTypeOfSymbol(symbol: Symbol, node: Node) {
             let type = getTypeOfSymbol(symbol);
             // Only narrow when symbol is variable of type any or an object, union, or type parameter type
             if (node && symbol.flags & SymbolFlags.Variable) {
                 if (isTypeAny(type) || type.flags & (TypeFlags.ObjectType | TypeFlags.Union | TypeFlags.TypeParameter)) {
+                    // `firstNarrowedTypeFromIfStatement` is only used when type is of union type.
+                    if (type.flags & TypeFlags.Union) {
+                        var firstNarrowedTypeFromIfStatement: Type;

```

Comment:
I'm getting some red squiggles if I define it as let. cannot find name `firNarroedTypeFromIfStatement` below.